From e6dbd09d2fdfa0d53917d4c4157b0329744c4fa2 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:51 -0400
Subject: [fs] btrfs: remove OPT_acl parse when acl disabled

Message-id: <1407179109-13714-345-git-send-email-sandeen@redhat.com>
Patchwork-id: 87237
O-Subject: [RHEL7.1 PATCH 344/422] Btrfs: remove OPT_acl parse when acl disabled
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Guangliang Zhao <lucienchao@gmail.com>

commit 45ff35d6b9f301cbfb6b1703466ebe4431c7567f
Author: Guangliang Zhao <lucienchao@gmail.com>
Date:   Mon May 12 11:04:33 2014 +0800

    Btrfs: remove OPT_acl parse when acl disabled

    Even CONFIG_BTRFS_FS_POSIX_ACL is not defined, the acl still could
    been enabled using a mount option, and now fs/btrfs/acl.o is not
    built, so the mount options will appear to be supported but will
    be silently ignored.

    Signed-off-by: Guangliang Zhao <lucienchao@gmail.com>
    Reviewed-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index 1b91578..89e530d 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -580,8 +580,15 @@ int btrfs_parse_options(struct btrfs_root *root, char *options)
    }
    break;
   case Opt_acl:
+#ifdef CONFIG_BTRFS_FS_POSIX_ACL
    root->fs_info->sb->s_flags |= MS_POSIXACL;
    break;
+#else
+   btrfs_err(root->fs_info,
+    "support for ACL not compiled in!");
+   ret = -EINVAL;
+   goto out;
+#endif
   case Opt_noacl:
    root->fs_info->sb->s_flags &= ~MS_POSIXACL;
    break;
-- 
1.7.1